scanner.c

```diff
@@ -38,34 +75,108 @@ scanner_clear(struct scanner_statistics * data) {
 	memset(data, 0, sizeof * data);
 }
 
+int get_next_field(char * buf, char ** line) {
+  int buf_i = 0;
+  // skip the previous tab
+  (*line)++;
+  while (**line != '\t') {
+    if (**line == '\0' || **line == '\n') return 0;
+    buf[buf_i] = **line;
+    (*line)++;
+    buf_i++;
+    if (buf_i == BUF_LEN) return 0;
+  }
+  buf[buf_i] = '/0';
+  return buf_i; 
+}
+
 static
 void
 scanner_process(const char * line, struct scanner_statistics * data) {
-	if (strstr(line, "Clear")) {
+	char buf[BUF_LEN];
+	int line_i = 0;
+  int sc_stat=-1;
+  int cc_stat=-1;
+
+  // Skip date
+	while (*line != '\t') {
+		if (*line == '\0' || *line == '\n') return;
+		line++;
+	}
+
+ 	// Load scan status
+	if (! get_next_field(buf, &line)) return;
```

Comment:
What exactly do you try to fix with this function? It just copies the content of `line` to `buf`. Why do you do the reference `&line` if you dereferenc all the time in `get_next_field` function. You should not bypass `const char *`, it is there from some reason.